ERCS/erc-7779.md

```diff
@@ -0,0 +1,191 @@
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+```solidity
+interface IInteroperableDelegatedAccount {
+
+	/*
+	 * @dev    Provides the namespace of the account.
+	 *         namespace of accounts can possibly include, account version, account name, wallet vendor name, etc
+	 * @notice this standard does not standardize the namespace format
+	 * e.g.,   "v0.1.2.7702Account.WalletProjectA"
+	 */
+	function accountId() external view returns (string);
+	
+	/*
+	 * @dev    Externally shares the storage bases that has been used throughout the account.
+	 *         Majority of 7702 accounts will have their distinctive storage base to reduce the chance of storage collision.
+	 *         This allows the external entities to know what the storage base is of the account.
+	 *         Wallets willing to redelegate already-delegated accounts should call accountStorageBase() to check if it confirms with the account it plans to redelegate.
+	 *
+	 *         The bytes32 array should be stored at the storage slot: keccak(keccak('InteroperableDelegatedAccount.ERC.Storage')-1) & ~0xff
+	 *         This is an append-only array so newly redelegated accounts should not overwrite the storage at this slot, but just append their base to the array.
+	 *         This append operation should be done during the initialization of the account.
+	 */
+	function accountStorageBases() external view returns (bytes32[]);
+
+}
+```
+
+```solidity
+interface IRedelegableDelegatedAccount {
+
+	/*
+	 * @dev    Function called before redelegation.
+	 *         This function should prepare the account for a delegation to a different implementation.
+	 *         This function could be triggered by the new wallet that wants to redelegate an already delegated EOA.
+	 *         It should uninitialize storages if needed and execute wallet-specific logic to prepare for redelegation.
+	 *         msg.sender should be the owner of the account.
+	 */
+	function onRedelegation() external returns (bool);
+
+}
+```
+
+Accounts MUST implement the `IInteroperableDelegatedAccount` to be compliant with the standard.
+
+Accounts MAY implement the `IRedelegableDelegatedAccount`.
+
+### `accountId()`
+
+This function is a view function to fetch the account information.
+
+A use case for this could be wallet showing the redelegation process e.g., Are you willing to migrate your account `“Wallet A” → “Wallet B”`.
+
+Wallet A information could be extracted from `accountId()`.
+
+### `accountStorageBases()`
+
+This function returns the list of base storage slots of that account has used.
+
+7702 Accounts do plan to use a custom non-zero storage slot to avoid storage collision as much as possible, however, there hasn’t been a standardized approach on how to fetch them.
```

Comment:
```suggestion
EIP-7702 Accounts do plan to use a custom non-zero storage slot to avoid storage collision as much as possible, however, there hasn’t been a standardized approach on how to fetch them.
```